src/main/scala/org/ensime/core/EnsimeEvent.scala

```diff
@@ -1,31 +1,71 @@
 package org.ensime.core
 
+import java.io.File
 import org.ensime.model.LineSourcePosition
-import org.ensime.util.NoteList
+import org.ensime.util.Note
 
-/**
- * A asynchronous swank protocol event
- */
+/** Asynchronous swank protocol event */
 sealed trait EnsimeEvent
-
 sealed trait GeneralSwankEvent extends EnsimeEvent
+sealed trait DebugEvent extends EnsimeEvent
 
+/** Generic background notification. */
 case class SendBackgroundMessageEvent(code: Int, detail: Option[String]) extends GeneralSwankEvent
+
+/** The presentation compiler is ready to accept requests. */
 case object AnalyzerReadyEvent extends GeneralSwankEvent
+
+/** The presentation compiler has finished analysing the entire project. */
```

Comment:
yeah hopefully the API module will consist of only case classes with docs on them and therefore document all the various protocols.